### PR TITLE
bug/tr-2909/solar-ignite-broken-result-outcome

### PR DESCRIPTION
**Pull request by poyuki**
# [TR-2909](https://oat-sa.atlassian.net/browse/TR-2909)
## Summary
Added result item label extraction from test items for remote delivery
## How to test
* install current brunch  for `extension-tao-outcomeui` on `tao` in `nextgen-stack`
* configure launching deliver on solar with lti in terre
    *  [Conflence doc](https://oat-sa.atlassian.net/wiki/spaces/~575513335/pages/1667204040/LTI+1.3+-+Deliver+Launch)
    * [Readme from `extension-tao-deliver-connect`](https://github.com/oat-sa/extension-tao-deliver-connect/blob/master/README.md)
* Publish delivery and execute
* Label items will be present on result page instead of `unknown`
---
Example of .env
```
FEATURE_FLAG_LTI1P3=true
DELIVERTENANT_COUNT=1
DELIVERTENANT_0_ID=1
DELIVERTENANT_0_LABEL="Customer 1 - Deliver - LTI 1.3 - .env"
DELIVERTENANT_0_CUSTOMER_ID=1
DELIVERTENANT_0_ROOT_URL=http://deliver.docker.localhost/api/v1/
DELIVERTENANT_0_OAUTH_CREDENTIALS_COUNT=1
DELIVERTENANT_0_OAUTH_CREDENTIALS_0_KEY=bbb1
DELIVERTENANT_0_OAUTH_CREDENTIALS_0_SECRET=bbb1
DELIVERTENANT_0_OAUTH_CREDENTIALS_0_ROLES_0=Learner
DELIVERTENANT_0_OAUTH_CREDENTIALS_0_ROLES_1=ContentDeveloper
DELIVERTENANT_0_OAUTH_CREDENTIALS_0_ROLES_COUNT=2
DELIVERTENANT_0_LTI_CREDENTIALS_0_KEY=aaa1
DELIVERTENANT_0_LTI_CREDENTIALS_0_SECRET=aaa1
DELIVERTENANT_0_LTI_CREDENTIALS_0_ROLES_COUNT=2
DELIVERTENANT_0_LTI_CREDENTIALS_0_ROLES_0=Learner
DELIVERTENANT_0_LTI_CREDENTIALS_0_ROLES_1=ContentDeveloper
DELIVERTENANT_0_LTI_CREDENTIALS_COUNT=1
```

### CODE REVIEW

File: model/ResultsService.php [hunk 1]
```diff
@@ -396,6 +397,34 @@ public function getItemFromItemResult($itemCallId, $itemVariables = [])
         return $item;
     }
 
+    /**
+     * @deprecated temp bugfix for remote deliveries
```
Comment (boajer) [inline model/ResultsService.php:401]: Could you refer to the proper solution while adding something deprecated?

Comment (wazelin) [reply model/ResultsService.php:401]: This should be a TODO with a follow up task IMO.

Comment (poyuki) [reply model/ResultsService.php:401]: may be it's not correct to mark it as deprecated. I just want to note that it's temp solution which waiting development of new features for this use case. May be you can suggest better way for this note, or may be it's shouldn't be noted?

Comment (poyuki) [reply model/ResultsService.php:401]: `TODO` sounds good. I will change it now

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by boajer — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ]  New code is respecting code style rules
- [ ]  New code is respecting best practices
- [ ]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [ ]  Acceptance criteria are respected
- [ ]  Pull request title and description are meaningful
- [ ]  Pull request's target is not master
- [ ]  Commits are following conventional commits
- [ ]  Commits messages are meaningful
- [ ]  Commits are atomic
- [ ]  Changelog is updated according to changes (if applicable)
- [ ]  Documentation is updated according to changes (if applicable)

**Review by emgolubev — APPROVED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Pull request merged**